### PR TITLE
Drop support for EOL Python 3.4

### PR DESCRIPTION
**Pull request by hugovk**
Python 3.4 is EOL and no longer receiving security updates (or any updates) from the core Python team.

Version | Release date | Supported until
--- | ---------- | ----------
2.7 | 2010-07-03 | 2020-01-01
3.4 | 2014-03-16 | 2019-03-16

Source: https://en.wikipedia.org/wiki/CPython#Version_history

It's also little used. Here's the pip installs for bump2version from PyPI for June 2019:

| category | percent | downloads |
|----------|--------:|----------:|
| 3.6      |  42.42% |     5,873 |
| 3.7      |  32.22% |     4,460 |
| 2.7      |  15.17% |     2,100 |
| 3.5      |   6.06% |       839 |
| 3.4      |   2.14% |       296 |
| null     |   1.54% |       213 |
| 3.8      |   0.39% |        54 |
| 3.3      |   0.07% |         9 |
| Total    |         |    13,844 |

Source: `pypistats python_minor bump2version --last-month # pip install pypistats`

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -27,6 +27,7 @@
             'bump2version = bumpversion.cli:main',
         ]
     },
+    python_requires='>=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*',
```
Comment (levrik) [inline setup.py]: I'm quite new to Python but couldn't that be expressed as `>=2.7, <3.0, >=3.5`?

Comment (ekohl) [reply setup.py]: https://www.python.org/dev/peps/pep-0440/#version-specifiers is a bit vague in this regard. The syntax used here in setup.py appears to be quite common from what I've seen. Perhaps this is just people copying each other though :)

Comment (hugovk) [reply setup.py]:
```python
    python_requires=">=2.7, <3.0, >=3.5",
```
Running `pip install -e .` on this with Python 3.7 gives:

`ERROR: Package 'foo' requires a different Python: 3.7.4 not in '>=2.7, <3.0, >=3.5'`

Comment (levrik) [reply setup.py]: @hugovk Yep. I saw that the `,` is basically a `and`.

Comment (hugovk) [reply setup.py]: It's a little bit ugly, all the more reason to drop almost-EOL Python 2.7 and write `python_requires=">=3.5"`!

https://python2woop.pw
https://python3statement.org

**Pull request merged**